### PR TITLE
actually really fix download_url end point for perl releases

### PR DESCRIPTION
**Pull request by haarg**

### CODE REVIEW

File: lib/MetaCPAN/Query/Release.pm [hunk 1]
```diff
@@ -1176,10 +1189,17 @@ sub find_download_url {
...
-    };
+    my $source_name = delete $source->{name};
+    if ( !$module_filter ) {
+        $source->{release} = $source_name;
+    }
+
+    my $module
+        = $hit->{inner_hits}{module}
+        ? $hit->{inner_hits}{module}{hits}{hits}[0]{_source}
+        : {};
```
Comment (atoomic) [inline lib/MetaCPAN/Query/Release.pm:1200]: probably always want to add a `$module //= {}` as we are going to dereference it later

File: lib/MetaCPAN/Query/Release.pm [hunk 2]
```diff
@@ -1176,10 +1189,17 @@ sub find_download_url {
...
+
+    return +{ %$source, %$module, @checksums, };
```
Comment (atoomic) [inline lib/MetaCPAN/Query/Release.pm:1202]: note that `$source` can be `undef`

Comment (haarg) [reply lib/MetaCPAN/Query/Release.pm:1202]: At this point, we have verified that there was a result in `$hit`. We are requesting specific `_source` fields. Even if none of those fields are available, `_source` will still exist unless Elasticsearch is entirely broken. `$source` can not be `undef`.

File: lib/MetaCPAN/Query/Release.pm (hunk 1 above)
Comment (haarg) [reply lib/MetaCPAN/Query/Release.pm:1200]: The ES query that would produce `{inner_hits}{module}` requires a module with a matching name. If `inner_hits` exists, `{inner_hits}{module}{hits}{hits}[0]{_source}` will always exist.

**Review by oalders — APPROVED** [review]

**Pull request merged**